multishard_mutation_query.cc

```diff
@@ -654,6 +654,16 @@ future<page_consume_result<ResultBuilder>> read_page(
     try {
         auto [ckey, result] = co_await query::consume_page(reader, compaction_state, cmd.slice, std::move(result_builder), cmd.get_row_limit(),
                 cmd.partition_limit, cmd.timestamp, *cmd.max_result_size);
+        const auto& cstats = compaction_state->stats();
+        tracing::trace(trace_state, "Page stats: {} partition(s), {} static row(s) ({} live, {} dead), {} clustering row(s) ({} live, {} dead) and {} range tombstone(s)",
+                cstats.partitions,
```

Comment:
For completeness, we should also count collection tombstones (detect large collections with few live cells). Can be done later.

We could also detect large collections by reporting max collection size. Maybe even log the exact column location into the tracing session on encountering.

Reply:
Good ideas but I'd prefer to do them as follow-ups as they (maybe) require reaching into `row::compact_and_expire()` internals, something I didn't have to do in this series.

I'll send a v2 addressing the other suggestions.